### ISSUE TITLE
How is BeyondDrive implemented

### ISSUE THREAD

**Issue by Python-Eric** [wjl2244/MeanFuser#11]
navsim/agents/meanfuser/meanfuser_model.py

`if self.use_beyonddrive is not None: 
            negative_index = (targets['negative_trajectory'].sum(-1).sum(-1) > 0)
            delta_negative_trajectory = diff_traj(targets['negative_trajectory'])
            negative_loss = ((predictions['diff_trajectory']-delta_negative_trajectory).abs().mean(-1).mean(-1))[negative_index].mean()
            negative_loss = -negative_loss
        else:
            negative_loss = 0.0`
How are negative trajectory samples obtained? Are they generated by data augmentation or simulation-based approaches? The loss function appears to adopt reward shaping, which is similar to DPO (Direct Preference Optimization). Is this correct?

**Comment by wjl2244** [comment]
Thank you for your questions, the BeyondDrive paper and code will be publicly available next weekend.

**Comment by wjl2244** [comment]
> navsim/agents/meanfuser/meanfuser_model.py
> 
> `if self.use_beyonddrive is not None: negative_index = (targets['negative_trajectory'].sum(-1).sum(-1) > 0) delta_negative_trajectory = diff_traj(targets['negative_trajectory']) negative_loss = ((predictions['diff_trajectory']-delta_negative_trajectory).abs().mean(-1).mean(-1))[negative_index].mean() negative_loss = -negative_loss else: negative_loss = 0.0` How are negative trajectory samples obtained? Are they generated by data augmentation or simulation-based approaches? The loss function appears to adopt reward shaping, which is similar to DPO (Direct Preference Optimization). Is this correct?

We have released [BeyondDrive](https://github.com/wjl2244/BeyondDrive).
